### DISCUSSION TITLE
Tracing the source of nautical depth data

### DISCUSSION THREAD

**Discussion by rptb1** [osmandapp/OsmAnd#12502] [Ideas]
I was directed to post here by OSMAnd support, because my conversation with them ended with a suggestion.

Summary: I tried to trace the source of the nautical depth data but was unsuccessful. My suggestion is that the OSMAnd app documents (via permalinks) the source of its data in a way that would allow it to be reproduced. This suggestion could apply to all map data.

Here are the edited highlights of the conversation with support.

I wrote:

> I have been trying to find out what the nautical depths provided by OSMAnd mean. I have read <https://osmand.net/features/nautical-charts> and I use both the "Nautical depth contours Europe contours" and "Nautical depth contours Europe points" maps. I have spent some time searching through the OpenSeaMap wiki, but with no luck.
>
> On a nautical chart, such as for example <https://www.visitmyharbour.com/harbours/solent/gosport-marinas/chart/F19A60E53A4B1/portsmouth-gosport> the depths are relative to a "chart datum" which is usually the lowest astronomical tide for the local port. That means that a "zero" depth on the chart is somewhere that is at sea level at the lowest possible local tide. It also means that the depths aren't consistent between charts if compared to, e.g., the WGS84 Geoid and the national datum.
>
> I compared the depths shown by OSMAnd around Portsmouth Harbour with those on the Admiralty chart, and found that they were between 1.8m and 11m deeper.
>
> I'm not expecting OSMAnd to show me perfect quality data suitable for navigation, of course! But whoever compiled the depths shown by OSMAnd must have some idea of their reference datum, and their reference geoid, and I can't seem to find a connection between your UI or your documentation and that information.

Support wrote:

> Right now OsmAnd compiled data from various open sources which might be far from perfect.
https://github.com/osmandapp/OsmAnd/blob/master/LICENSE#L132
>
> So, we're going to recompile this year and try to find new sources of information.

I wrote:

> May I suggest that, as well as finding better information sources, you also find an accessible way to document the sources?
>
> It could be something simple. For example, on the map downloads list you currently only have names and descriptions for the maps. Perhaps you could also have a persistent link that leads to further information on the web, for people who need to know more details.
>
> Ideally, the link would allow anyone to reproduce the data -- e.g. a git hash to the data or a procedure that reproduces it, so that there's a permanent record.

**Comment by vshcherb** [comment]
Yes some source provided in https://github.com/osmandapp/OsmAnd/blob/master/LICENSE#L132. 
All our sources are compiled and manually crafted i.e. there was some manual work done in QGis to compile sources, I'm not sure if we distribute these results publicly or not.  

Main person behind this data generation @xmd5a2 is currently away probably he could enlight more information how it was done.

**Comment by quantenschaum** [comment]
I have been asking myself the same question. I looked into the depth data for the Netherlands. It was impossible to find information about the depth datum (LAT/NAP), where the data actually comes from and how up to date it is. A comparison with current and official data revealed that the NL depth data is complete rubbish, it is probably very old and should not be used at. It's a proof of concept of depth data in OsmAnd IMHO.

**Comment by quantenschaum** [comment]
- How could I prepare my own depth data file?
- How is depth data entered into OSM?

**Reply by vshcherb** [reply]
If you have OSM XML file with contours ,you can create OBF file using OsmAndMapCreator. It's possible to convert Shapefile to OSM XML using ogrogr utility

**Reply by quantenschaum** [reply]
Cool! Is there some documentation out there that describes how the tags in the OSM XML should look like?

**Reply by quantenschaum** [reply]
How do I have to tag depth contours and spot soundings that they appear on the map? I cannot find any docs on this topic.

**Reply by quantenschaum** [reply]
If I tag them with `contour=depth` and `depth=12` nothing is shown on the map.

**Reply by vshcherb** [reply]
Did you check with inspector? https://docs.osmand.net/docs/technical/map-creation/create-offline-maps-yourself - check rendering_types.xml parameter it documents which tags are indexed for rendering. https://docs.osmand.net/docs/technical/map-creation/how-to-inspect-an-obf

**Reply by quantenschaum** [reply]
Thanks! I found [this](https://wiki.openstreetmap.org/wiki/OsmAndMapCreator), your link is helpful as well.

I did download some OSM data with JOSM, added some points and lines and ran the OMC. It worked. I can import the OBF into OsmAnd and the buoys I added are shown on the map, but *not* the contour lines and points.

The docs do not tell much, it's more trying to find the right sections and trial and error. :disappointed: 

From [`rendering_types.xml`](https://github.com/osmandapp/OsmAnd-resources/blob/master/obf_creation/rendering_types.xml)I would guess that tagging with `contour=depth, depth=10` or `point=depth depth=10` would do the trick, but it does not work.

There are depth data available for download from within OsmAnd. How were these OFBs created?

**Comment by quantenschaum** [comment]
Just to mention it: 

Tags needed for contour lines
- `contour=depth`
- `depth=<value in m>`
- `contourtype=[5m,10m,20m,50m,100m]`

Tags needed for spot soundings
- `point=depth`
- `depth=<value in m>`

**Reply by quantenschaum** [reply]
see also https://github.com/osmandapp/OsmAnd/discussions/18116
